Feature/eco counter

jankosk opened the pull request:
Initial version for eco counters. There are still some missing Swedish translations and workarounds for missing data e.g. counts data for 'Oulu_kapy' and 'Oulu_ELY' sites.

Commit:
Add initial eco counters on the map

Commit:
Add initial eco counter popup

Commit:
Add initial EcoCounterContent component

Commit:
WIP add chart

Commit:
Enhance LineChart to work with multiple data sets

Also query both user type directions in EcoCounterContent

Commit:
Add buttons to change userType for EcoCounterContent

Commit:
Enhance Eco Counter graph

Also update moment.locale after changing language

Commit:
Add SelectEcoCounterRow

jankosk commented on app/component/map/tile-layer/TileLayerContainer.js:
```diff
@@ -45,13 +46,23 @@ import TileContainer from './TileContainer';
+const getSiteIdFromChannelId = (id, domain) => {
+  if (domain === 'Oulu_kaupunki') {
+    return `${id.slice(0, 2)}0${id.slice(3, id.length)}`;
+  }
```
I'm assuming here that the third character of the`siteId` in 'Oulu_kaupunki' domains always denotes the different direction and user types, and that 0 is always the "base" site. I'm not sure if this is always the case. Also do the other domain siteIds work similarly?

kryysler replied:
There is no guarantee that this is the case. The "user type", i.e. pedestrian or bicycle is contained in it's own property `userType` for each channel. The direction however is not specified directly anywhere.

kryysler replied:
After reviewing the documentation at http://eco-test2.com/apidoc/wso2/apidoc.html, it says the `sens` property includes the direction; "Direction detection information. {1: IN direction, 2: OUT direction, 5: No direction distinction}".

jankosk commented on app/component/map/popups/EcoCounterPopup.js:
```diff
@@ -0,0 +1,162 @@
+              new EcoCounterDataRoute({
+                outId: channels[0].siteId,
+                inId: channels[1].siteId,
+                domain: ecoCounterSite.domain,
```
I'm assuming here that the first one in channels array for is "out" direction and the second one is "in" direction. Not sure if this is the case.

kryysler replied:
See comment above.

Commit:
Remove unnecessary ecoCounterSite query

Use map api to get eco counter sites userTypes and directions.

Commit:
Enhance eco counter error handling

Also fix eco counter rows selection

Commit:
Handle null direction in eco counters

Commit:
Add Swedish translations for eco counter

jankosk commented:
Added missing translations and removed unnecessary ecoCounterSites query. It should now work even if the direction is not specified and if only walking or cycling data is available.

Commit:
Add unit tests for EcoCounterContent
